test/integration/controller-manager/cr-validation_test.go

```diff
@@ -0,0 +1,126 @@
+
+package cmtest
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/klog/v2/ktesting"
+	kastesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
+	"k8s.io/kubernetes/test/integration/framework"
+	"sigs.k8s.io/yaml"
+
+	ksapi "github.com/kubestellar/kubestellar/api/control/v1alpha1"
+	kscrd "github.com/kubestellar/kubestellar/pkg/crd"
+)
+
+func TestBindingPolicyValidation(t *testing.T) {
+	testWriter := framework.NewTBWriter(t)
+	logger, ctx := ktesting.NewTestContext(t)
+	logger.Info("Starting etcd server")
+	framework.StartEtcd(t, testWriter)
+	logger.Info("Starting TestController")
+	t.Log("Beginning TestController")
+	ctx, cancel := context.WithCancel(ctx)
+	testServer, err := kastesting.StartTestServer(t, kastesting.NewDefaultTestServerOptions(), []string{}, framework.SharedEtcd())
+	if err != nil {
+		t.Fatalf("Failed to kastesting.StartTestServer: %s", err)
+	}
+	fullTeardwon := func() {
+		cancel()
+		testServer.TearDownFn()
+	}
+	t.Cleanup(fullTeardwon)
+	config := testServer.ClientConfig
+	if err != nil {
+		t.Fatalf("Failed to create Kubernetes client: %s", err)
+	}
+	logger.Info("Started test server", "config", config)
+	configCopy := *config
+	config4json := &configCopy
+	config4json.ContentType = "application/json"
+	logger.Info("REST config for JSON marshaling", "config", config4json)
+	apiextClient, err := apiextensionsclientset.NewForConfig(config)
+	if err != nil {
+		t.Fatalf("Failed to create apiextensions client: %s", err)
+	}
+	err = kscrd.ApplyCRDs(ctx, "test", apiextClient.ApiextensionsV1().CustomResourceDefinitions(), logger)
+	if err != nil {
+		t.Fatalf("Failed to apply KubeStellar CRDs: %s", err)
+	}
+	bpCRD, err := apiextClient.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, "bindingpolicies.control.kubestellar.io", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to fetch BindingPolicy CRD: %s", err)
+	}
+	bpYAML, err := yaml.Marshal(bpCRD)
+	if err != nil {
+		t.Fatalf("Failed to marshal BindingPolicy CRD: %s", err)
+	}
+	t.Log("Fetched BindingPolicy CRD:\n" + string(bpYAML))
+	dynClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		t.Fatalf("Failed to create dynamic client: %s", err)
+	}
+	dynIfc := dynClient.Resource(ksapi.SchemeGroupVersion.WithResource("bindingpolicies"))
+	converter := runtime.DefaultUnstructuredConverter
+	for _, testCase := range []struct {
+		name     string
+		specJSON string
+		expectOK bool
+	}{
+		{name: "junk-field-in-spec", specJSON: `{"junk": 1}`, expectOK: false},
+		{name: "junk-field-in-policy", specJSON: `{"downsync": [{"junq": true}]}`},
+		{name: "no-test-in-policy", specJSON: `{"downsync": [{"createOnly": true, "statusCollection": {"statusCollectors": ["phred"]}}]}`},
```

Comment:
Need some help to understand: Why no `expectOK` for these two test cases?

Reply:
It is not necessary to specify every field in a struct constructor. You can omit a field when its zero value is the value you want.

Reply:
I see. It might be more consistent with line 92 (the first test case) --- not a blocker.